app/views/snacks/show.html.erb

```diff
@@ -0,0 +1,13 @@
+<h1><%= @snack.name %></h1>
+<p>Snacks:</p>
+<p>Price: <%= number_to_currency(@snack.price) %></p>
+<p>Locations:</p>
+<ul id="snack-locations-<%= @snack.id %>">
+  <% @snack.machines.each do |machine| %>
+    <li>
+      <%= machine.location  %>, 
+      <%= machine.snacks.count %> kinds of snacks, 
```

Comment:
This will work (machine.snacks.count), but I'd move this to a model method so that you can take away the logic that  `.count` does out of the view.